CMakeLists.txt

```diff
@@ -1,80 +1,39 @@
-	find_program(FIND_GIT git)
-	if("${FIND_GIT}" STREQUAL "FIND_GIT-NOTFOUND")
-		message(FATAL_ERROR "Could not find 'git' please set GCF_GIT_PATH:STRING")
-	else()
-		set(GCF_GIT_PATH ${FIND_GIT})
-		message(STATUS "Found: ${GCF_GIT_PATH}")
-	endif()
-else()
-	if(NOT EXISTS ${GCF_GIT_PATH})
-		message(WARNING "Could not find git: ${GCF_GIT_PATH}")
-	else()
-		message(STATUS "Found: ${GCF_GIT_PATH}")
-	endif()
-endif()
-
-if("${GCF_PYTHON_PATH}" STREQUAL "NOTSET")
-	find_program(FIND_PYTHON ${GCF_PYTHON_NAME})
-	if("${FIND_PYTHON}" STREQUAL "FIND_PYTHON-NOTFOUND")
-		message(FATAL_ERROR "Could not find '${GCF_PYTHON_NAME}' please set GCF_PYTHON_PATH:STRING")
-	else()
-		set(GCF_PYTHON_PATH ${FIND_PYTHON})
-		message(STATUS "Found: ${GCF_PYTHON_PATH}")
-	endif()
-else()
-	if(NOT EXISTS ${GCF_PYTHON_PATH})
-		message(WARNING "Could not find python: ${GCF_PYTHON_PATH}")
-	else()
-		message(STATUS "Found: ${GCF_PYTHON_PATH}")
-	endif()
-endif()
-
-if("${GCF_CLANGFORMAT_PATH}" STREQUAL "NOTSET")
-	find_program(FIND_CLANGFORMAT clang-format)
-	if("${FIND_CLANGFORMAT}" STREQUAL "FIND_CLANGFORMAT-NOTFOUND")
-		message(FATAL_ERROR "Could not find 'clang-format' please set GCF_CLANGFORMAT_PATH:STRING")
-	else()
-		set(GCF_CLANGFORMAT_PATH ${FIND_CLANGFORMAT})
-		message(STATUS "Found: ${GCF_CLANGFORMAT_PATH}")
-	endif()
-else()
-	if(NOT EXISTS ${GCF_CLANGFORMAT_PATH})
-		message(WARNING "Could not find clang-format: ${GCF_CLANGFORMAT_PATH}")
-	else()
-		message(STATUS "Found: ${GCF_CLANGFORMAT_PATH}")
-	endif()
-endif()
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR})
+find_package(ClangFormat REQUIRED)
 
+set(GCF_CLANGFORMAT_STYLE "file" CACHE STRING
+	"Parameter pass to clang-format -style=<here>")
+set(GCF_IGNORE_LIST "" CACHE STRING
+	"Semi colon separated list of directories to ignore")
 set(GCF_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/git-cmake-format.py)
-string(REGEX REPLACE "\\\\" "/" GCF_GIT_PATH ${GCF_GIT_PATH})
-string(REGEX REPLACE "\\\\" "/" GCF_PYTHON_PATH ${GCF_PYTHON_PATH})
-string(REGEX REPLACE "\\\\" "/" GCF_CLANGFORMAT_PATH ${GCF_CLANGFORMAT_PATH})
 
-execute_process(COMMAND git rev-parse --show-toplevel
+execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --show-toplevel
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
   OUTPUT_VARIABLE GCF_GIT_ROOT
   OUTPUT_STRIP_TRAILING_WHITESPACE)
 if(NOT GCF_GIT_ROOT)
-  message(WARNING "Not in a git repository")
+  message(SEND_ERROR "Not in a git repository")
+else()
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/git-pre-commit-hook
+    ${GCF_GIT_ROOT}/.git/hooks/pre-commit
+    @ONLY
+  )
+  unset(GCF_GIT_ROOT)
+
+  add_custom_target(format
+    ${PYTHON_EXECUTABLE} ${GCF_SCRIPT}
+    --cmake ${GIT_EXECUTABLE}
+    ${CLANG_FORMAT_EXECUTABLE} -style=${GCF_CLANGFORMAT_STYLE} -ignore=${GCF_IGNORE_LIST}
+	  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  )
+  unset(GCF_SCRIPT)
 else()
-  message(STATUS "Found git root: ${GCF_GIT_ROOT}")
+  message(SEND_ERROR "Cannot locate top .git repository")
```

Comment:
I don't want this to stop generation of build scripts, please change to `WARNING`.